engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -1123,7 +1137,7 @@ public long getNextSlaveSequenceValue( String slaveSequenceName, long incrementV
     }
   }
 
-  public SlaveServer getClient() {
+  public synchronized SlaveServer getClient() {
```

Comment:
a Read Lock (if necessary) on the variables makes more sense than synchronizing the entire method.